controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -88,57 +91,66 @@ public String getName() {
-                                                                                                    log.debug(context.getRequestId(), "table for kvts created {}", kvtsTableName);
-                                                                                                    if (e != null) {
-                                                                                                        throw new CompletionException(e);
-                                                                                                    }
-                                                                                                })),
-                                               getReaderGroupsInScopeTableName(context)
-                                                       .thenCompose(rgTableName -> storeHelper.createTable(rgTableName, context.getRequestId())
-                                                                                              .thenAccept(v -> {
-                                                                                                  log.debug(context.getRequestId(), "table for reader groups created {}", rgTableName);
-                                                                                                  if (e != null) {
-                                                                                                      throw new CompletionException(e);
-                                                                                                  }
-                                                                                              })),
-                                               getAllStreamTagsInScopeTableNames(context)
-                                                       .thenCompose(tagTableNames -> {
-                                                           List<CompletableFuture<Void>> futures = new ArrayList<>();
-                                                           for (String tableName : tagTableNames) {
-                                                               futures.add(storeHelper.createTable(tableName, context.getRequestId()));
-                                                           }
-                                                           return Futures.allOf(futures);
-                                                       })
-                                                       .thenAccept(v -> {
-                                                           if (e != null) {
-                                                               throw new CompletionException(e);
-                                                           }
-                                                       }));
+        return checkScopeExistsInTable(scopeName, context).thenCompose(exist -> {
+            if (exist) {
+                throw new ScopeDeletionInProgressException("Scope deletion in progress");
```

Comment:
No exceptions please. this needs to be handled using return status.

Reply:
This method returns CompletableFuture<Void>, so we can not return return the status from here